### PR TITLE
Migrate to the new standard schema

### LINKED ISSUE #559: Charter Africa | Test Airtable New Airtable Schema
There's an updated airtable schema in the ecosystem where social media columns are part of table columns as opposed to the current one which has a separate table for Social Media.

### PR DESCRIPTION
**Pull request by koechkevin**
## Description

This PR updates charter africa to use new standard schema.
As per discussion and test of the originally implemented schema, the difference in  time/size is negligible. 

Fixes #559 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by kilemensi** [comment]
Is this a _test_ or a **migration** to the new schema @koechkevin ?

**Comment by koechkevin** [comment]
> Is this a _test_ or a **migration** to the new schema @koechkevin ?

This is a migration.

File: apps/charterafrica/src/lib/ecosystem/airtable/processData.js [hunk 1]
```diff
@@ -47,27 +47,32 @@ function mapSupporterIdsToObjects(supporterIds, config, { partnersData }) {
...
+    getData("facebook"),
+    getData("youtube"),
+    getData("slack"),
+    getData("tiktok"),
+    getData("linkedIn"),
+    getData("instagram"),
+    getData("telegram"),
+    getData("discord"),
+    getData("whatsapp"),
+  ].filter(Boolean);
```
Comment (kilemensi) [inline apps/charterafrica/src/lib/ecosystem/airtable/processData.js:72]: We can declare supported social media platforms somewhere at the top of the file
```js
const SUPPORTED_SOCIAL_MEDIA = "twitter,facebook,youtube,slack,tiktok,linkedIn,instagram,telegram,discord,whatsapp";
```

We can also use array if you prefer that.

```suggestion
  return SUPPORTED_SOCIAL_MEDIA.split(",").map(getData).filter(Boolean);
```

File: apps/charterafrica/src/lib/ecosystem/airtable/processData.js [hunk 2]
```diff
@@ -47,27 +47,32 @@ function mapSupporterIdsToObjects(supporterIds, config, { partnersData }) {
...
-  const {
-    schema: { socialMediaTableColumns },
-  } = config;
-  const { name, url } = socialMediaTableColumns;
-  const { socialMediaData = [] } = tableData;
-  const mapped = socialMedia.map((id) => {
-    const { fields } = socialMediaData.find((item) => item.id === id) || {};
-
-    if (!fields) {
+function mapSocialMediaColumnsToObjects(columns, data) {
```
Comment (kilemensi) [inline apps/charterafrica/src/lib/ecosystem/airtable/processData.js:50]: Are we mapping to _objects_?

File: apps/charterafrica/src/payload/fields/ecosystem/socialMediaFields.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      }),
+      airtableColumnSelect({
+        schema,
+        tableField,
+        overrides: {
+          name: "whatsapp",
+          label: "Whatsapp",
+        },
+      }),
+    ],
```
Comment (kilemensi) [inline apps/charterafrica/src/payload/fields/ecosystem/socialMediaFields.js:81]:
```suggestion
    ],
    admin: {
      hideGutter: true,
    },
```

We need to review how the CMS _looks_ as well.
Screenshot 2023-09-12 at 09 35 16

**Review by kilemensi — COMMENTED** [review]
👍🏽 

---

Only done a quick review, I'm sure the rest of the team will do a more thorough review.

I've also noticed two files: `airtable.old.js` and `ecostystem.old.js` in the `lib/ecosystem` folder. Are they still being used or they are just on my local machine.

File: apps/charterafrica/src/lib/ecosystem/airtable/processData.js (hunk 2 above)
Comment (koechkevin) [reply apps/charterafrica/src/lib/ecosystem/airtable/processData.js:50]: We ar mapping to array of objects

**Comment by kelvinkipruto** [comment]
@koechkevin When updating the Contributors' Table, Telegram social media does not appear in the dropdown list.

**Review by kelvinkipruto — APPROVED** [review]

**Comment by koechkevin** [comment]
> @koechkevin When updating the Contributors' Table, Telegram social media does not appear in the dropdown list.

@kelvinkipruto then the column does not exist in airtable. Not a code problem

**Review by kilemensi — APPROVED** [review]

**Pull request merged**